borderless/explainer.md

```diff
@@ -0,0 +1,512 @@
+
+## Goals
+
+- Enable full control over the appearance of the title bar area
+- Ensure draggable regions work with borderless mode
+
+## Non-goals
+
+- Implementation of
+  [additional windowing controls](https://github.com/ivansandrk/additional-windowing-controls/blob/main/awc-explainer.md),
+  which is another closely-related project
+- Making borderless mode dynamic (with e.g. JavaScript APIs or something else)
+- Changing display mode in app settings (e.g. from borderless to standalone)
+- Support for mobileOSs/macOS/windows
+- Support for non-IWAs (Isolated Web Apps)
+
+## Proposal
+
+The solution proposed consists of the following parts:
+
+1. A new `display_override` option `borderless` for the web app manifest.
+2. Enabling draggable regions when in borderless mode.
+3. CSS display-mode media query for borderless.
+4. Leveraging Window Management API permission (_earlier “Multi-screen Windows
+   Placement” API permission_) for borderless mode.
+5. Check that the app is Isolated Web App.
+
+### 1. Setting the `display_override` to borderless
+
+To provide the maximum addressable area for web content, the User Agent (UA)
+will create a frameless window removing all UA provided Chrome, leaving only
+resizing of the window from its borders. The removed window controls will be
+enabled using AWC and HTML/JavaScript/CSS (see non-goals and
+[additional-windowing-controls explainer](<[http://go/additional-windowing-controls](https://github.com/ivansandrk/additional-windowing-controls/blob/main/awc-explainer.md)>)).
+Additionally to hiding the native title bar, also the menu button is hidden.
+
+Example apps in borderless mode could look e.g. like below but the appearance
+would eventually fully depend on what the developer would implement.
+
+![Example borderless PWA](./images/borderless-app-with-AWC.png)
+
+The desire to remove the title bar will be declared within the web app manifest
+by setting the `display_override` to `borderless`. The display_override value
+will be ignored on unsupported OSs and they will follow the normal display mode
+hierarchy.
+
+```
+{
+  "display_override": [ "borderless" ]
+}
+```
+
+### 2. Enabling draggable regions
+
+When the app enters the borderless mode,
+[draggable regions](https://github.com/WICG/window-controls-overlay/blob/main/explainer.md#defining-draggable-regions-in-web-content)
+`-webkit-app-region: drag;` will be enabled.
+
+### 3. CSS display-mode media query for borderless
+
+Similarly to other display modes, borderless should be queryable with @media.
+
+Example media query with borderless:
+
+```
+@media (display-mode: borderless) {
+  .any-css-class-name {
+    margin: 5px;
+  }
+}
+```
+
+See:
+[https://developer.mozilla.org/en-US/docs/Web/CSS/@media/display-mode](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/display-mode)
+
+### 4. Leveraging Window Management permission for borderless
+
+In order for the borderless mode to activate, the app needs to have Window
```

Comment:
It makes me uncomfortable tying these two permissions together.

Borderless is a much scarier permission than management, due to the spoofing risk. I would not want every app I grant management permission to to magically get borderless permission as well.

Reply:
The permission design has been discussed and approved by security. I think the differentiation here is that it's only supported for isolated web apps, so providing the window management permission on its own would still not enable the feature.